index.js

```diff
@@ -0,0 +1,59 @@
+const bodyParser = require('body-parser')
+const express = require('express')
+
+const teams = require('./teams.json')
+
+const app = express()
+
+app.get('/teams', (request, response) => {
+  response.send(teams)
+})
+
+app.get('/teams/:input', (request, response) => {
+  const requestInfo = request.params.input
+  const teamByParams = teams.filter(team => {
+    return (
+      team.id === Number(requestInfo) ||
+      team.abbreviation.toUpperCase() === requestInfo.toUpperCase()
+    )
+  })
+  return teamByParams.length
+    ? response.send(teamByParams)
+    : response.sendStatus(404)
+})
+app.use(bodyParser.json())
+app.post('/teams', (request, response) => {
+  let {
+    id,
+    location,
+    mascot,
+    abbreviation,
+    conference,
+    division
+  } = request.body
```

Comment:
I would not include `id` here if you are going to set it to a value for them. Would annoy the user to have to supply an unused value.